src/rust/src/file_reader.rs

```diff
@@ -34,11 +34,19 @@ pub struct FeatureIter<R, S> {
     /// File offset within feature section
     cur_pos: u64,
     /// All features read or end of file reached
```

Comment:
Should be smth. like "Reading state" then.

I feel a temptation to include `cur_pos` in the Reading state, but this would the code probably make less readable.

Reply:
> Should be smth. like "Reading state" then.

Whoops good catch!

> I feel a temptation to include cur_pos in the Reading state, but this would probably make the code less readable.

I've attempted this here:
https://github.com/michaelkirk/flatgeobuf/pull/new/mkirk/gh-361-unexpected-eof-cur-pos-in-state

But I think I agree that it's slightly less readable. Or at least not obviously better.